velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,261 @@
+    auto offsets = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+    auto sizes = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+
+    // Create a vector of maps with 1 key , 1 value
+    vector_size_t offset = 0;
+    for (vector_size_t i = 0; i < size; i++) {
+      rawSizes[i] = 1;
+      rawOffsets[i] = offset;
+      offset += 1;
+    }
+
+    return std::make_shared<MapVector>(
+        pool_.get(),
+        MAP(BIGINT(), ARRAY(BIGINT())),
+        nullptr,
+        size,
+        offsets,
+        sizes,
+        mapKeys,
+        arrayValues);
+  }
+
+ private:
+  BufferPtr bufferPtr_;
+};
+
+TEST_F(PrestoHasherTest, integralTypes) {
+  integralTest<int8_t>();
+  integralTest<int16_t>();
+  integralTest<int32_t>();
+  integralTest<int64_t>();
+  assertHash<int32_t>({99999}, {-6111805494244633133});
+  assertHash<int64_t>({99999}, {-6111805494244633133});
+
+  // Test timestamp
+  assertHash<Timestamp>(
+      {Timestamp(1, 100),
+       Timestamp(10, 10),
+       Timestamp(100, 1000),
+       Timestamp(0, 0),
+       std::nullopt},
+      {2343331593029422743, -3897102175227929705, 3043507167507853989, 0, 0});
+
+  // Test Date
+  assertHash<Date>(
+      {Date(0), Date(1000), std::nullopt}, {0, 2343331593029422743, 0});
+}
+
+TEST_F(PrestoHasherTest, doubleType) {
+  assertHash<double>(
+      {1.0, 0.0, std::nan("0"), 99999.99},
+      {2156309669339463680L, 0, -6389168865844396032, 9140591727513491554});
+}
+
+TEST_F(PrestoHasherTest, floatType) {
+  assertHash<float>(
+      {1.0f, 0.0f, std::nanf("0"), 99999.99f},
+      {-6641611864725600567L, 0, 6018425597037803642, -6781949276808836923});
+}
+
+TEST_F(PrestoHasherTest, VarcharType) {
+  assertHash<StringView>(
+      {StringView("abcd"),
+       StringView(""),
+       std::nullopt,
+       StringView(u8"Thanks \u0020\u007F")},
+      {-2449070131962342708, -1205034819632174695, 0, 2911531567394159200});
+}
+
+TEST_F(PrestoHasherTest, boolType) {
+  assertHash<bool>({true, false}, {1231, 1237});
+}
+
+TEST_F(PrestoHasherTest, arrays) {
+  auto O = [](std::vector<std::optional<int64_t>> data) {
+    return std::make_optional(data);
+  };
+
+  auto baseArrayVector = vectorMaker_.arrayVectorNullable<int64_t>(
+      {O({1, 2}),
+       O({3, 4}),
+       O({4, 5}),
+       O({6, 7}),
+       O({8, 9}),
+       O({10, 11}),
+       O({12, std::nullopt})});
+
+  auto arrayExpected = std::vector<int64_t>(
+      {4329740752828760473,
+       655643799837771513,
+       8633635089947142034,
+       9138382565482297209,
+       1065928229506940121,
+       8616704993676952121,
+       1942639070761256766});
+
+  assertHash(arrayExpected, baseArrayVector);
+
+  // Array of arrays.
+  auto arrayOfArrayVector =
+      vectorMaker_.arrayVector({0, 2, 4}, baseArrayVector);
+
+  auto arrayofArraysExpected = std::vector<int64_t>(
```

Comment:
typo: arrayOf..., but it would be easier to read if you simply inlined that variable